CMakeLists.txt

```diff
@@ -41,9 +41,7 @@ endif()
 # Set a default build type if none was specified
 # See https://blog.kitware.com/cmake-and-the-default-build-type/ for details.
 set(default_build_type "RelWithDebInfo")
-if(EXISTS "${CMAKE_SOURCE_DIR}/.git" AND NOT WIN32)
-  # Debug builds are broken on Windows due to CMake/vcpkg linking issues.
-  # As a workaround, we just default to RelWithDebInfo on Windows.
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
```

Comment:
Why reintroducing this change again, after it was removed in #3977. This generates a unusable slow Windows build by default.
Debug dependencies are needed only in rare cases, when you need to step line by line through the dependency code using a debugger.

Reply:
Yes I have considered that as well.
But I have finally removed it because the reasoning is now different. I will revert that and update the comments.